worlds/dkc3/Client.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import asyncio
+
+from NetUtils import ClientStatus, color
+from SNIClient import Context, snes_buffered_write, snes_flush_writes, snes_read
+from Patch import GAME_DKC3
+
+snes_logger = logging.getLogger("SNES")
+
+# DKC3 - DKC3_TODO: Check these values
+ROM_START = 0x000000
+WRAM_START = 0xF50000
+WRAM_SIZE = 0x20000
+SRAM_START = 0xE00000
+
+SAVEDATA_START = WRAM_START + 0xF000
+SAVEDATA_SIZE = 0x500
+
+DKC3_ROMNAME_START = 0x00FFC0
+DKC3_ROMHASH_START = 0x7FC0
+ROMNAME_SIZE = 0x15
+ROMHASH_SIZE = 0x15
+
+DKC3_RECV_PROGRESS_ADDR = WRAM_START + 0x632     # DKC3_TODO: Find a permanent home for this
+DKC3_FILE_NAME_ADDR = WRAM_START + 0x5D9
+DEATH_LINK_ACTIVE_ADDR = DKC3_ROMNAME_START + 0x15     # DKC3_TODO: Find a permanent home for this
+
+
+async def deathlink_kill_player(ctx: Context):
+    pass
+    #if ctx.game == GAME_DKC3:
+        # DKC3_TODO: Handle Receiving Deathlink
+
+
+async def dkc3_rom_init(ctx: Context):
+    if not ctx.rom:
+        ctx.finished_game = False
+        ctx.death_link_allow_survive = False
+        game_name = await snes_read(ctx, DKC3_ROMNAME_START, 0x15)
+        if game_name is None or game_name != b"DONKEY KONG COUNTRY 3":
+            return False
+        else:
+            ctx.game = GAME_DKC3
+            ctx.items_handling = 0b111  # remote items
+
+        rom = await snes_read(ctx, DKC3_ROMHASH_START, ROMHASH_SIZE)
+        if rom is None or rom == bytes([0] * ROMHASH_SIZE):
+            return False
+
+        ctx.rom = rom
+
+        #death_link = await snes_read(ctx, DEATH_LINK_ACTIVE_ADDR, 1)
+        ## DKC3_TODO: Handle Deathlink
+        #if death_link:
+        #    ctx.allow_collect = bool(death_link[0] & 0b100)
+        #    await ctx.update_death_link(bool(death_link[0] & 0b1))
+    return True
+
+
+async def dkc3_game_watcher(ctx: Context):
+    if ctx.game == GAME_DKC3:
```

Comment:
One of the features that's missing from SNIClient is "unsetting" ctx.game. Breaking out of the game loop when you load a different rom would help in avoiding bugs.

Reply:
As discussed over Discord, this will be pursued in a separate PR more directly targeted at SNIClient upgrades